noir-projects/noir-contracts/contracts/ecdsa_public_key_note/src/lib.nr

```diff
@@ -3,13 +3,13 @@ use dep::aztec::prelude::{NoteHeader, NoteInterface, NullifiableNote, PrivateCon
 use dep::aztec::{
     note::utils::compute_note_hash_for_nullify, keys::getters::{get_nsk_app, get_public_keys},
     protocol_types::{address::AztecAddress, constants::GENERATOR_INDEX__NOTE_NULLIFIER, hash::poseidon2_hash_with_separator},
-    macros::notes::note_custom_interface, generators::Ga1 as Gx_1, generators::Ga2 as Gx_2,
-    generators::Ga3 as Gy_1, generators::Ga4 as Gy_2, generators::Ga5 as G_owner, generators::G_slot
+    macros::notes::note_custom_interface
 };
 
 use std::hash::from_field_unsafe;
 
 global ECDSA_PUBLIC_KEY_NOTE_LEN: u32 = 5;
+global NOTE_HASH_SEPARATOR: u32 = 1744684982; // Arbitrary generator index
```

Comment:
This is a bit dangerous, I think, in the sense that it sets a dangerous precedent, and could lead to users getting used to arbitrary generators. Then a malicious contract deployer could provide a malicious generator and users might not check.
There is a dedicated epic for domain separation, but perhaps we should have a way to derive a `u32` separator by hashing a string `"EcdsaPublicKeyNote"` at compile time, somehow?

Reply:
You have a point. Replaced its use with `GENERATOR_INDEX__NOTE_HASH` which was already in `constants.nr`. I checked and GENERATOR_INDEX__NOTE_HASH was not used for anything now.